packages/common/src/hooks/useDocument/useDocument.ts

```diff
@@ -89,6 +89,12 @@ export const useDocument = <
   }, [data]);
 
   const { draft } = state;
+  const draftRef = useRef(draft);
+  draftRef.current = draft;
 
-  return { state, draft, save: mutateAsync, dispatch };
+  const save = async () => {
+    mutateAsync(draftRef.current);
+  };
```

Comment:
instead of passing the draft in which I thought would be best, instead making a ref so that we can just call `save()` and the most up to date draft is saved (dispatches are async and so was causing me troubles as react waits till the end of an event handler to fire the dispatches, so it can batch them into a single re-render)

Reply:
wow! nifty